sdk/platform/noplatform/src/az_noplatform.c

```diff
@@ -8,23 +8,3 @@
 AZ_NODISCARD int64_t az_platform_clock_msec() { return 0; }
 
 void az_platform_sleep_msec(int32_t milliseconds) { (void)milliseconds; }
-
```

Comment:
We still need these

Reply:
based on @antkmsft , we don't right?  I will resolve this.  Jeff, please let me know if otherwise

Reply:
Correct, we can get rid of all mutex stuff in the PAL. Anton & I came up with  much better plan - he'll add what we need after this PR is merged.